crates/weaver_resolved_schema/src/body.rs

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(rustdoc::invalid_html_tags)]
+
+//! Specification of a resolved body field.
+
+use crate::attribute::AttributeRef;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use weaver_semconv::attribute::{AttributeType, Examples, RequirementLevel};
+use weaver_semconv::body::{BodyFieldSpec, BodySpec};
+use weaver_semconv::stability::Stability;
+
+/// An attribute definition.
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Hash, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct Body {
+    /// The body specification.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub fields: Option<Vec<BodyField>>,
+    // Not yet defined in the spec or implemented in the resolver
+    // The body value when there are no fields
+    // #[serde(skip_serializing_if = "Option::is_none")]
+    // pub value: Option<Value>
+}
+
+/// An attribute definition.
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Hash, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct BodyField {
+    /// Attribute name.
+    pub name: String,
+    /// A reference to an attribute definition, used to populate the relevant
+    /// fields of the body field, unless they are overridden by the body field.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub r#attr: Option<AttributeRef>,
```

Comment:
Good point, I originally added as a just in case. But you are correct this is not required and should not be exposed on the resolved schema -- will remove.

Reply:
Removed as the `ref` / `alias` option is now also removed